src/components/program-progress/data/hooks.jsx

```diff
@@ -46,3 +50,23 @@ export function useLearnerProgramsListData(enterpriseUuid) {
   }, [enterpriseUuid]);
   return [camelCaseObject(learnerProgramsListData), fetchError];
 }
+
+export function useHasLicenseOrCoupon() {
+  const {
+    subscriptionPlan,
+    subscriptionLicense: userSubscriptionLicense,
```

Comment:
Interesting in hearing your take here, I generally avoid calling `useContext` in custom hooks and prefer passing in the arguments instead. I find it easier to reason about and test. This could also probably be a regular helper function.

Reply:
Hmm, yeah I could see a valid argument for both ways here. Agreed passing the arguments instead does make the hook most testable, but from the usage point of view, I could also argue abstracting where those arguments come from could make sense if the caller of the hook doesn't already have that data extracted from the context. It's nice to not have to worry about context values as a user of `useHasLicenseOrCoupon`. But, it does make testing less obvious, too since you'd still have to know the internals of what context providers are necessary when testing the parent consumer component regardless.

B/c this hook is only used in one place, I will revert it back to treating it as a regular helper function as originally intended and pass in the required context arguments instead. I probably wouldn't refactor for this change though in this PR if it were used in multiple places.